### ISSUE TITLE
terms having a wrong sort Id

### ISSUE THREAD

**Issue by daejunpark** [kframework/k-legacy#2138]
In kool-static, a term has a wrong sort during execution. @andreistefanescu @cos do you have any idea?

For example, a term `_._(class(Id(#"Main")),, Id(#"Main"))` has a sort `Id`, which is not correct. Because of this, the following rule is being applied over and over again, which leads to non-terminating.

```
685   rule <k> X:Id => this . X ...</k>
686        <tenv> Rho </tenv>
687     when notBool(X in keys(Rho))
```

To reproduce:

On the top of `fix/ac-bug` branch 27d1eea93b,
`kompile kool-typed-static.k`, and `krun` the following program:

```
class Main {
  void Main(){
  }
}
```

After the initial 6 steps, you will find the term ever growing, that keeps adding the prefix `this .` in front of the term, as follows:

```
class(Id(#"Main")),, Id(#"Main")
this(.KList),, _._(class(Id(#"Main")),, Id(#"Main"))
this(.KList),, _._(this(.KList),, _._(class(Id(#"Main")),, Id(#"Main")))
this(.KList),, _._(this(.KList),, _._(this(.KList),, _._(class(Id(#"Main")),, Id(#"Main"))))
...
```

**Comment by daejunpark** [comment]
related to https://github.com/kframework/k/pull/2130#issuecomment-180397690

**Comment by grosu** [comment]
Yes, it is definitely wrong for `Id.Main` to be an `Id`.  Maybe the regex corresponding to `Id`s is what needs to change?

Thanks, @daejunpark, for putting your finger on the bug.

**Comment by cos** [comment]
The problem is the following:
By [subsorting rule](https://github.com/kframework/k/blob/fix/ac-bug/k-distribution/tutorial/2_languages/2_kool/2_typed/2_static/kool-typed-static.k#L692) any `_._` is an LValue:

```
// We would like to say the following instead of the predicate rule below:
//  syntax LValue ::= Exp "." Id
// but unfortunately that currently generates some ambiguous parsings.
  rule isLValue(_:Exp . _:Id) => true
```

but `_._` is also an `Exp`, and the only `Exp` that we know of that is also an `LValue` is `Id` (i.e., `Id` is the unique GLB of `Exp` and `LValue`) so the inference algorithm concludes that `_._` must be an `Id`.

**Comment by cos** [comment]
I'll try to solve it by the split:

```
syntax FieldReference ::= Exp "." Id   [strict(1)]
syntax Exp ::= FieldReference
```

Let's see...

**Comment by cos** [comment]
Also named the `ArrayReference` and `constructors.kool` type-checks on my computer. Let's see if the rest are passing.

**Comment by cos** [comment]
The sort issue seems to be solved. 

@grosu, just making sure I understand this correctly, a subsort pre-regularity check would have caught the above problem at compile time, right?
